src/schemas/inscription.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from 'zod';
+
+export const CharmSchema = z.enum([
+  'burned',
+  'coin',
+  'cursed',
+  'epic',
+  'legendary',
+  'lost',
+  'mythic',
+  'nineball',
+  'palindrome',
+  'rare',
+  'reinscription',
+  'unbound',
+  'uncommon',
+  'vindicated',
+]);
+
+export const InscriptionSchema = z.object({
+  address: z.string().nullable(),
+  charms: z.array(CharmSchema),
+  children_count: z.number().int().nonnegative().nullable().optional(),
+  children: z.array(z.string()),
+  content_length: z.number().int().nonnegative(),
+  content_type: z.string(),
+  effective_content_type: z.string(),
+  fee: z.number().int().nonnegative(),
+  height: z.number().int().nonnegative(),
+  id: z.string(),
+  next: z.string().nullable(),
+  number: z.number().int().nonnegative(),
+  parents: z.array(z.string()),
+  previous: z.string().nullable(),
+  rune: z.string().nullable(),
+  sat: z.number().int().nonnegative().nullable(),
+  satpoint: z.string(),
+  timestamp: z.number().int(),
+  value: z.number().int().nonnegative(),
+  metaprotocol: z.string().nullable().optional(),
```

Comment:
get rid of all `.optional()` and only use nullable.

Check other types as well.